### ISSUE TITLE
Reading list view broken in mobile

### ISSUE THREAD

**Issue by Rafi993** [forem/forem#11983]
**Describe the bug**

Reading list view is not responsive in mobile

**To Reproduce**

1. Go to https://dev.to/readinglist
2. Reading list layout wont be responsive 

**Expected behavior**
 Readinglist should be responsive in mobile

**Screenshots**

![Screenshot_20201220-132914_Brave](https://user-images.githubusercontent.com/8092120/102708288-e2725180-42c7-11eb-9550-06023dc60911.png)

**Smartphone**

- Device: readmi note 6 pro
- OS, version: 10
- Browser, version: 1.17.75

**Comment by maestromac** [comment]
I ran into this issue too but I can scroll it. I think the problem is that the sidebar is given too much room. Because I follow a lot of tags, I have to scroll really far down to see my reading list articles.

**Comment by fdoxyz** [comment]
I think it would be great if we make a decision here that could benefit other mobile sized screens as well. There's [this issue](https://github.com/forem/forem/issues/7769) that also needs to handle a "left sidebar" that doesn't render well in mobile.

A long vertical list with items doesn't work for mobile in these situations and the solution implemented in `/notifications` is something that could be reused on any screen with this similar layout (filter/control in left sidebar + main content). Here is the comparison of how the `/notifications` [left sidebar](https://github.com/forem/forem/blob/master/app/views/notifications/index.html.erb#L49-L51) works compared to the reading list: [Video link](https://share.getcloudapp.com/9Zu0bzbY)

In the Reading list we have a [Preact component](https://github.com/forem/forem/blob/master/app/javascript/readingList/readingList.jsx) in place, and a high level summary of the logic implemented is a "multiple selection" (selecting/unselecting tags as filters). A horizontal scroll of the tags could be an interesting option to consider. Since this is not a unique option to be selected, like in `/notifications`, a `select` wouldn't work. I don't think we'll need to tweak the Preact component but i'm not 100% sure this will be the case.

The reason I mention all this is to share as much context as possible. I would love to hear what @forem/frontend and @forem/design think about this

**Comment by aitchiss** [comment]
I think as it is at the moment, it's really not clear to a mobile user that those tags are there to help filter the reading list, and I totally agree it's too much to scroll through to actually reach your articles.

I think an idea would be to have a "filter" dropdown containing each of the tags and a checkbox for if you want to include it in the filter. I think this a fairly common pattern and would allow us to tuck away all those tags in the mobile view. I feel like a horizontal scroll could work but is maybe a bit less user friendly for such a long list

**Comment by nickytonline** [comment]
> I think as it is at the moment, it's really not clear to a mobile user that those tags are there to help filter the reading list, and I totally agree it's too much to scroll through to actually reach your articles.
> 
> I think an idea would be to have a "filter" dropdown containing each of the tags and a checkbox for if you want to include it in the filter. I think this a fairly common pattern and would allow us to tuck away all those tags in the mobile view. I feel like a horizontal scroll could work but is maybe a bit less user friendly for such a long list

Just an idea, but if the filter used a [datalist](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist), it would drill down so as to not make the dropdown too overwhelming.

**Comment by aitchiss** [comment]
> Just an idea, but if the filter used a [datalist](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist), it would drill down so as to not make the dropdown too overwhelming.

Nice! Only downside to that is I _think_ (would love to be wrong though) with a datalist we'd lose the ability to select multiple. Maybe another good use-case for something like Downshift though.

**Comment by nickytonline** [comment]
> > Just an idea, but if the filter used a [datalist](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist), it would drill down so as to not make the dropdown too overwhelming.
> 
> Nice! Only downside to that is I _think_ (would love to be wrong though) with a datalist we'd lose the ability to select multiple. Maybe another good use-case for something like Downshift though.

Yeah, it's single selection. In that case, we'd need a select with multi-select enabled. Perhaps product design has some ideas about this. cc: @ludwiczakpawel @lisasy .

**Comment by ludwiczakpawel** [comment]
We've actually chatted about it earlier, in [this issue](https://github.com/forem/forem/issues/11033) as well.

TL;DR we hypothesize that "**multiselecting** tags" isn't really useful for anyone - instead user should be able to filter reading list by selecting **only one** tag. With that in mind, we could easily replace classic list that we have on desktop with `<select>` field that would work on mobile - we do that in other places right now as well (see /settings navigation or notifications filtering). With that in mind, _this navigation_ would need a bit of refactoring I believe.

**Comment by vaidehijoshi** [comment]
See relevant potential solution here: https://github.com/forem/forem/issues/11033#issuecomment-716680053
